tools/pkglint/lib/rules.ts

```diff
@@ -1616,13 +1616,68 @@ export class UbergenPackageVisibility extends ValidationRule {
   }
 }
 
+/**
+ * No experimental dependencies.
+ * In v2 all experimental modules will be released separately from aws-cdk-lib. This means that:
+ * 1. Stable modules can't depend on experimental modules as it will creates a cyclic dependency.
+ * 2. Experimental modules shouldn't depend on experimental modules as it will create a coupling between their graduation (cause of 1).
+ * 2 specify "shouldn't" as in some cases we might allow it (using the `excludedDependencies` map), but the default is to not allow it.
+ */
+export class NoExperimentalDependents extends ValidationRule {
+  public name = 'no-experimental-dependencies';
+
+  // experimental -> experimental dependencies that are allowed for now.
+  private readonly excludedDependencies = new Map([
+    ['@aws-cdk/aws-secretsmanager', ['@aws-cdk/aws-sam']],
+    ['@aws-cdk/aws-kinesisanalytics-flink', ['@aws-cdk/aws-kinesisanalytics']],
+    ['@aws-cdk/aws-apigatewayv2-integrations', ['@aws-cdk/aws-apigatewayv2']],
+    ['@aws-cdk/aws-apigatewayv2-authorizers', ['@aws-cdk/aws-apigatewayv2']],
+    ['@aws-cdk/aws-events-targets', ['@aws-cdk/aws-kinesisfirehose']],
+  ]);
+
+  private readonly excludedModules = ['@aws-cdk/cloudformation-include'];
+
+  public validate(pkg: PackageJson): void {
+    if (this.excludedModules.includes(pkg.packageName)) {
+      return;
+    }
+    if (!isCdkModuleName(pkg.packageName)) {
+      return;
+    }
+
+    if (!isIncludedInMonolith(pkg)) {
+      return;
+    }
+
+    Object.keys(pkg.dependencies).forEach(dep => {
+      if (!isCdkModuleName(dep)) {
+        return;
+      }
+
+      const shortName = dep.split('/')[1];
+      // eslint-disable-next-line @typescript-eslint/no-require-imports
+      const stability = require(path.resolve('../', shortName, 'package.json')).stability;
```

Comment:
Please don't write these kinds of rules that rely on physical locations of files in the source repository.

Instead, if you must find a dependency's `package.json`, do something like this:

```ts
const dependencyPj = path.resolve(`${shortName}/package.json`, { paths: [searchPath] });
const packageJson = await readJson(dependencyPj);
```

Reply:
So! We only need to read the `package.json` file of dependencies, which means they were already loaded during the build phase => we can require the dependency name, and we don't need a path:

```ts
const stability = require(`${dep}/package.json`).stability;
```